x-pack/plugin/otel-data/src/main/resources/component-templates/metrics-otel@mappings.yaml

```diff
@@ -0,0 +1,55 @@
+version: ${xpack.oteldata.template.version}
+_meta:
+  description: Default mappings for the logs index template installed by x-pack
+  managed: true
+template:
+  mappings:
+    properties:
+      data_stream.type:
+        type: constant_keyword
+        value: metrics
+      start_time:
+        type: date_nanos
+      metrics:
+        type: passthrough
+        dynamic: true
+        priority: 1
+      unit:
+        type: keyword
+        time_series_dimension: true
+        ignore_above: 1024
+    dynamic_templates:
```

Comment:
Let's move these dynamic templates to `metrics@mappings`, so that we can have a simpler logic for shippers so that they can use the same dynamic templates regardless of which metric data stream is targeted.

Reply:
Based on a quick check, it seems `time_series_metric` gets ignored when targeting a non tsdb data stream.